### PR TITLE
chore(ci): build and publish flatpak on release

### PR DESCRIPTION
**Pull request by airtonix**
Lampray will now build and publish flatpaks.

Furthermore to facilitate hallway testing, when you comment in a PR `/build` a PRs can have a flatpak generated as a artifact (not a release) and you'll see a comment link to the created preview flatpak.

This is useful for:

- the release PRs to get a preview of what a release is going to be like.
- fix and feature PRs: test if your changes are helpful.

## Todo

- [ ] test and ensure that the preview workflow only runs when the chatops command is used.
- [ ] rename the flatpak id for previews so it's different from the release build
- [ ] work out how publishing flatpaks work: what detaisl are required etc.

### CODE REVIEW

File: docs/building-from-source.md
Comment (airtonix) [inline docs/building-from-source.md:1]: woops, i need to discard these changes.

File: .github/workflows/release.yml
Comment (airtonix) [inline .github/workflows/release.yml:1]: in light of https://github.com/CHollingworth/Lampray/pull/122, this artifact upload needs to change to uploading several files: 

- the binary
- the icons

File: com.github.chollingworth.Lampray.yml
Comment (airtonix) [inline com.github.chollingworth.Lampray.yml:1]: https://github.com/CHollingworth/Lampray/pull/122

- need to add more file sources for the desktop file and icons

**Comment by airtonix** [comment]
I'll wait for #122 to get merged with any references to icons before moving ahead with this one.

**Comment by airtonix** [comment]
@CHollingworth ok retargeted. 

next steps: 

1. merge this to the tmp feature branch.
2. @Plarpoon will take over ensuring that we can make a flatpak build. We'll post  examples flatpaks.
3. once we see that we have working flatpak builds, we'll be able to continue working on getting github actions to build and release the flatpak release.

**Comment by CHollingworth** [comment]
Apologies for the wait.

**Pull request merged**